runtime/lua/vim/lsp/handlers.lua

```diff
@@ -321,13 +321,14 @@ end
 --see: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_hover
 M['textDocument/hover'] = M.hover
 
----@private
 --- Jumps to a location. Used as a handler for multiple LSP methods.
----@param _ (not used)
 ---@param result (table) result of LSP method; a location or a list of locations.
 ---@param ctx (table) table containing the context of the request, including the method
 ---(`textDocument/definition` can return `Location` or `Location[]`
-local function location_handler(_, result, ctx, _)
+---@param config table Configuration table.
+---     - reuse_win: (default=nil) Jump to existing window if buffer is already open.
+function M.location_handler(_, result, ctx, config)
```

Comment:
I'm not sure if we still want to follow this model. For better multi client support for references and other locations we'd need to merge the results from all clients and then display them.  The location_handler signature wouldn't fit for that because the ctx with it's client_id implies a single client